File: game/scripts/vscripts/modifiers/upgrades/items/modifier_upgrade_item_blade_of_alacrity.lua
modifier_upgrade_item_blade_of_alacrity = class({})

function modifier_upgrade_item_blade_of_alacrity:RemoveOnDeath() return false end
function modifier_upgrade_item_blade_of_alacrity:IsPurgable() return false end

function modifier_upgrade_item_blade_of_alacrity:OnCreated()
    self.speed_buff_pct = 15
	self.as_speed = 50
end

function modifier_upgrade_item_blade_of_alacrity:OnRefresh()
    if IsServer() then
        self:IncrementStackCount()
    end
end

function modifier_upgrade_item_blade_of_alacrity:DeclareFunctions()
    return { 
		MODIFIER_PROPERTY_MOVESPEED_BONUS_PERCENTAGE,
        MODIFIER_PROPERTY_ATTACKSPEED_BONUS_CONSTANT,
    }
end

function modifier_upgrade_item_blade_of_alacrity:GetModifierAttackSpeedBonus_Constant()
	return self.as_speed
end

function modifier_upgrade_item_blade_of_alacrity:GetModifierMoveSpeedBonus_Percentage(params)
    return self.speed_buff_pct
end

function modifier_upgrade_item_blade_of_alacrity:GetTexture()
    return 'item_blade_of_alacrity'
end

File: src/vscripts/modifiers/upgrades/items/modifier_upgrade_item_blade_of_alacrity.ts
import { registerModifier } from "../../../lib/dota_ts_adapter";
import { ModifierUpgrade } from "../modifier_upgrade";

@registerModifier("modifier_upgrade_item_blade_of_alacrity")
class ModifierItemBladeOfAlacrity extends ModifierUpgrade {
    speedBuffPct = 15;
    asSpeed = 50;

    OnRefresh() {
        if (IsServer()) {
            this.IncrementStackCount();
        }
    }

    DeclareFunctions() {
        return [ModifierFunction.MOVESPEED_BONUS_PERCENTAGE, ModifierFunction.ATTACKSPEED_BONUS_CONSTANT];
    }

    GetModifierAttackSpeedBonus_Constant() {
        return this.asSpeed;
    }

    GetModifierMoveSpeedBonus_Percentage() {
        return this.speedBuffPct;
    }

    GetTexture() {
        return "item_blade_of_alacrity";
    }
}
